core/src/main/java/io/substrait/expression/FieldReference.java

```diff
@@ -149,7 +149,12 @@ public FieldReference constructOnExpression(Expression expr) {
 
     @Override
     public FieldReference constructOnRoot(Type type) {
```

Comment:
Let's modify this to take `Type.Struct`. That way it works cleanly with `constructOnRoot(myRel.getRecordType())`

Reply:
Done